perfkitbenchmarker/benchmark_sets.py

```diff
@@ -125,9 +130,18 @@ def GetBenchmarksFromFlags():
   If multiple sets or mixes of sets and benchmarks are specified, this will
   return the union of all sets and individual benchmarks.
   """
+  # Create a dictionary of valid benchmark names and modules
+  # TODO(voellm): I really think the benchmarks package init should
+  # build the dictionary
+  valid_benchmarks = {}
+  for benchmark_module in benchmarks.BENCHMARKS:
+    valid_benchmarks[benchmark_module.GetInfo()['name']] = benchmark_module
+
   benchmark_names = set()
   for benchmark in FLAGS.benchmarks:
-    if benchmark in BENCHMARK_SETS:
+    if benchmark == ALL_SET:
```

Comment:
There is one bug here... looks like the user would need to pass "all_set" vs "all".  The _set is assumed.